src/engine/server/server.cpp

```diff
@@ -1168,7 +1168,7 @@ void CServer::PumpNetwork()
 				CUnpacker Unpacker;
 				Unpacker.Reset((unsigned char*)Packet.m_pData+sizeof(SERVERBROWSE_GETINFO), Packet.m_DataSize-sizeof(SERVERBROWSE_GETINFO));
 				int SrvBrwsToken = Unpacker.GetInt();
-				if(Unpacker.Error())
+				if(Unpacker.Error() || SrvBrwsToken < 0)
 					continue;
 
 				CPacker Packer;
```

Comment:
```suggestion
				CPacker Packer;
				Packer.Reset();
				Packer.AddRaw(SERVERBROWSE_INFO, sizeof(SERVERBROWSE_INFO));
				Packer.AddInt(SrvBrwsToken);
```

Instead of disallowing negative `SrvBrwsToken`, I'd prefer to see the `GenerateServerInfo` function fixed. It's weird that its behavior differs depending on the passed `SrvBrwsToken`.

Could the `GenerateServerInfo` function be changed to `GenerateServerInfo(CPacker *pPacker, bool IncludeClientInfo)`, never resetting the passed `CPacker`?

Reply:
I can do that for Teeworlds' `GenerateServerInfo` function and DDNet's `CServer::GetServerInfoSixup` function, but I don't know how to do that cleanly for DDNet's `CServer::SendServerInfo` function, which packs the token in multiple chunks, if this should be consistent between all three functions.